scavenger-agent-java/src/main/java/com/navercorp/scavenger/javaagent/collecting/CodeBaseScanner.java

```diff
@@ -178,6 +184,11 @@ private boolean isAdditionalPackage(ClassNode clazz) {
         return additionalPackagePaths.stream().anyMatch(clazz.name::startsWith);
     }
 
+    private boolean isAdditionalByRegex(ClassNode clazz) {
+        String className = clazz.name.replaceAll("/", ".");
+        return additionalByRegex.stream().anyMatch(regex -> Pattern.matches(regex, className));
```

Comment:
In this approach, every time a class file is scanned, all regex strings in `additionalByRegex` and `excludeByRegex` will be recompiled to `Pattern`. Instead of holding them as `List<String>` in field, I think it would be better to pre-compile the regex patterns in constructor and store them as `List<Pattern>`.

> If a pattern is to be used multiple times, compiling it once and reusing it will be more efficient than invoking this method each time.
https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#matches-java.lang.String-java.lang.CharSequence-

Reply:
As mentioned, `bytebuddy.matcher.StringMatcher.Mode.MATCHES` uses `Pattern.matches`. 
Therefore, switching to a List<Pattern> would improve performance.
Resolved in commit 0f630a6ded43036568165878dd1e0777a3512a2b.